config/nrfconnect/chip-module/CMakeLists.txt

```diff
@@ -305,6 +306,19 @@ if (CONFIG_CHIP_LIB_SHELL)
     target_link_options(chip INTERFACE -Wl,--whole-archive -lCHIPShell -Wl,--no-whole-archive)
 endif()
 
+if (CONFIG_CHIP_MALLOC_SYS_HEAP)
+    target_link_options(chip INTERFACE
```

Comment:
I think this is really practical use case of the GCC wrap functionality, have not used it myself in practice. Maybe a good candidate for the quick knowledge sharing meeting :)